### DISCUSSION TITLE
purpose_of_use_details

### DISCUSSION THREAD

**Discussion by AMJohnsen** [NorskHelsenett/Tillitsrammeverk#60] [General]
Formålet med behandlingen av personopplysninger virker vel og merke ok, men hvordan har epj. systemene tenkt til å løse dette. For helsepersonell med 1:1 forhold til pasienten (eks. en spesialist) så vil formålet være enkelt å sette, men i en behandlerrelasjon der man ikke har samme 1:1 forhold så kan man begynne med et formål og underveis så kan formålet endre seg. Er det tenkt at helsepersonell underveis skal sette flere formål/endre formål? I verste fall så kan det ende opp med at man hardkoder inn et formål. 

hva slags betraktninger har man gjort seg i denne settingen @steinarnoem?

**Comment by troelde** [comment]
Jeg representerer DIPS som i dag benytter et kodeverk som er felles for alle DIPS-kundene. I hovedsak så er det systemet som setter koden automatisk hvis pasienten er i et pasientforløp (henvisning->venteliste->planlagt behandling->Poliklinisk/Innleggelse) på en avdeling hvor helsepersonell (leger, sykepleiere og kontor) jobber slik at systemet kan utlede denne koden. Vi kaller det for _implisitt beslutning_. Dersom pasienten ikke er i et pasientforløp, så må helsepersonellet begrunne tilgang manuelt ved å velge en kode og angi en  tekstlig begrunnelse (f.eks. henvendelse fra pasient, tilsyn, ..). Vi kaller det for _eksplisitt beslutning._  Kodeverket finnes [her](https://hl7norway.github.io/AuditEvent/currentbuild/CodeSystem-carerelation.html)

**Reply by AMJohnsen** [reply]
@troelde - hva tenker du vedr. andre fagsystemer som ikke setter koden automatisk? Kan det bety at de alltid må velge en kode og oppgi en tekstlig begrunnelse ved enhver tid? Det kan muligens være en barriere mtp brukeropplevelsen. Tidligere har jeg erfart at slike barrierer blir forbigått ved en hardkoding, og dersom ikke alle fagsystemer har samme funksjonalitet som DIPS, eller velger å utvikle samme funksjonalitet, så vil ikke purpose_of_use_details gi merverdi.

**Reply by troelde** [reply]
@AMJohnsen : Et fagsystem vil i mange tilfeller ikke ha all informasjon som skal til for å sette kode automatisk. Dette er typisk informasjon som lagres i pasientadministrative systemer knyttet til planlagt behandling. Er virksomheten stor nok (f.eks. Sykehus og kommune), er det naturlig at dette tilbys som en felles tjeneste som alle fagsystemer kan benytte (f.eks. slik som hos HSØ/SP) for å sjekke om pasienten er i et pasientforløp ved institusjonen og begrunnelse for tilgang (purpose_of_use_details). Ved mindre virksomheter, så vil det jo være enklere forhold som nesten alltid er knyttet til "pasientkonsultasjon", "prøvetakning", etc., slik at det er naturlig med "hardkoding" med mindre det er noe helt spesielt - f.eks. akutte situasjoner ("break the glass"). Da holder det kanskje med purpose_of_use , og ikke nødvendig å ha med detaljer i purpose_of_use_details. Siden jeg ikke kjenner forholdene i mindre virksomheter, så tenker jeg andre bør uttale seg her. 

PS! Hos HF der DIPS er i bruk, så er det purpose_of_use_details som vises i kolonnen "bakgrunn for tilgang" i tilgangsloggen i flisen "pasientjournal" i helsenorge.no. Den skal gi pasienten nødvendig kontekst for innsynet som ble gjort. Nå skal det sies at kodeverket nok kan forenkles, siden det er noen av kodene som ikke gir mening å vise til pasient.

**Comment by kyrrem** [comment]
Jeg jobber i Aidn som tilbyr EPJ-system for kommunesektoren. Jeg tenker at detaljeringsgrad her vil være et pågående arbeid, der det i første iterasjon for oss nok vil være hardkodet til et lite antall - i hvert fall i purpose_of_use, men der details kan muligens ha større spredning. Jeg tenker at disse ikke nødvendigvis må være i harmoni på tvers av systemer for å ha verdi, så lenge hensikten primært er å gi nok informasjon om årsak til å produsere et bra grunnlag for innsynslogg til helsenorge. Det viktigste blir kanskje da at vi har en felles forståelse av hvordan disse ulike feltene henger sammen med rapportering til helsenorge? Altså en god forståelse av hva som er tenkt å vises hvor til sluttbruker. 

Å utvide bruken til å feks autorisasjonslogikk basert på noe av dette virker ikke umiddelbart enkelt eller logisk for meg. Da vil nok heller rolle og organisasjon være steder der man kan forsøke å utøve mer presisisjon i tilgangskontrollen, kanskje i kombinasjon med tid / periode. I kommunal sammenheng er ikke ting som henvisning / diagnose etc like relevant for å angi scope på tilgang, ettersom de fleste PLO-tjenester må se hele pasienten i sammenheng, der svært mange av dem er eldre pasienter med potensielt mange samtidige  diagnoser og tildels kompleks medisinering.

Vi vil også forsøke i så stor grad som mulig å unngå popup med "velg årsak" etc, og heller ha litt dårligere presisjon i tilfeller der det er vanskelig å utlede automatisk. Nettopp pga brukeropplevelse som nevnt over.

Disclaimer: Jeg har ikke hatt anledning til å følge de siste diskusjonene i møtene her, så min forståelsen av feltene kan være mangelfull.

**Comment by kyrrem** [comment]
Vi jobber også nå i Aidn med å utvikle en tjeneste som kjenner til access reason /  purpose_of_use på tvers av Aidns delsystemer, denne kan etter hvert potensielt bli brukt av andre - også fagsystemer. Dette vil fungere nokså likt det som beskriver fra DIPS, der man kan benytte et gyldig token fra vår IDP til å få ut informasjon om tjenstlig behov etc.

**Comment by NEIsaksen** [comment]
![crpou](https://github.com/NorskHelsenett/Tillitsrammeverk/assets/125996363/190fe762-f45b-4c74-9538-ab9040a39d30)
Her: https://github.com/NorskHelsenett/Tillitsrammeverk/blob/main/specs/informasjons_og_datamodell.md
Så er dette attributtet obligatorisk det ene stedet og ikke det andre stedet i teksten. Er vel feil?
@steinarnoem
